src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -128,142 +149,145 @@ static TrackOptionsSet sTrackOptions = {
     { 0x1770, 4, 1, 6 },
 };
 
-s32 func_80BA8820(EnBaba* this, PlayState* play) {
-    this->unk_144 = (EnOssan*)SubS_FindActor(play, &this->unk_144->actor, ACTORCAT_NPC, ACTOR_EN_OSSAN);
+s32 EnBaba_FindBombShopkeeper(EnBaba* this, PlayState* play) {
+    //! The bomb shopkeeper is an EnSob1, but initalizes itself with id `ACTOR_EN_OSSAN`
+    //! Note if there are other `EnOssan` actors, it may find that instance instead
+    //! in which case `EnSob1` struct acceses would be incorrect
+    this->bombShopkeeper = (EnSob1*)SubS_FindActor(play, &this->bombShopkeeper->actor, ACTORCAT_NPC, ACTOR_EN_OSSAN);
 
-    if (this->unk_144 != NULL) {
+    if (this->bombShopkeeper != NULL) {
         return true;
     }
     return false;
 }
 
-void func_80BA886C(EnBaba* this, PlayState* play) {
+void EnBaba_HandleConversation(EnBaba* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
-    switch (this->unk_1E0) {
+    switch (this->textId) {
         case 0:
-            if (this->unk_40A & 8) {
+            if (this->stateFlags & BOMB_SHOP_LADY_STATE_AUTOTALK) {
                 if (gSaveContext.save.weekEventReg[33] & 8) {
-                    this->unk_1E0 = 0x2A34;
+                    this->textId = 0x2A34;
```

Comment:
Can you provide a short summary of the text? You can find a list of all text here: https://docs.google.com/spreadsheets/d/1cbFFzoKaCgqMybAfahhDYPHq1vaCo2pc8v6BfZKhk2M/edit#gid=1312253055